src/Plugin/VoteUpDownWidgetBase.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Drupal\vud\Plugin;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Base class for Vote Up/Down Widget plugins.
+ */
+abstract class VoteUpDownWidgetBase extends PluginBase implements VoteUpDownWidgetInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->getPluginDefinition()['label'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  function alterTemplateVars($widget_template, &$variables) {
+    $criteria = [
+      'entity_type' => $variables['entity_type'],
+      'entity_id' => $variables['entity_id'],
+      'value_type' => $variables['points'],
+      'tag' => $variables['tag'],
+    ];
+
+    // Find sum of the values
```

Comment:
Please follow [doc standards on inline comments](https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards#inline).
This change applies to several places on this commit, please fix them.